lib/utils.js

```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+function getNativeImagePath() {
+  if (process.platform === 'darwin') {
+    try {
+      return require('google-closure-compiler-osx');
+    } catch (e) {
+      return;
+    }
+  }
+  try {
+    return require('google-closure-compiler-linux');
+  } catch (e) {
+  }
+}
+
+function getFirstSupportedPlatform(platforms) {
+  const platform = platforms.find((platform, index) => {
+    switch (platform.toLowerCase()) {
+      case "java":
+        if (index === platforms.length - 1) {
+          return true;
+        }
+        return process.env.JAVA_HOME;
+
+      case "javascript":
+        return true;
+
+      case "native":
+        if (getNativeImagePath()) {
+          return true;
+        }
+    }
+  });
+  if (!platform) {
+    throw new Error('No supported platform for closure-compiler found.');
+  }
+  return platform;
+}
+
+/**
+ * Parse cli provided flags and create an object
+ *
+ * @param {!Array<string>} flags
+ * @return {!Object<string, (string|boolean|Array<string>)>}
+ */
+function parseCliFlags(flags) {
+  const flagShortcuts = new Map([
+    ['-O', 'compilation_level'],
+    ['-W', 'warning_level']
+  ]);
+
+  const compilerFlags = {};
+  let currentFlag;
+  const addFlagValue = flagValue => {
+    if (!currentFlag) {
+      currentFlag = 'js';
+    }
+
+    if (compilerFlags[currentFlag] && !Array.isArray(compilerFlags[currentFlag])) {
+      compilerFlags[currentFlag] = [compilerFlags[currentFlag], flagValue];
+    } else {
+      compilerFlags[currentFlag] = flagValue;
+    }
+    currentFlag = undefined;
+  };
+  flags.forEach(flagOrValue => {
+    let nextFlag;
+    let maybeFlag = flagOrValue;
+    let eqIndex = flagOrValue.indexOf('=');
+    if (eqIndex > 0) {
+      maybeFlag = maybeFlag.substr(0, eqIndex);
+    }
+    if (flagShortcuts.has(maybeFlag)) {
+      nextFlag = flagShortcuts.get(maybeFlag);
+      if (eqIndex > 0) {
+        nextFlag += flagOrValue.substr(eqIndex);
+      }
+    } else if (/^--/.test(maybeFlag)) {
```

Comment:
Should `--` by itself indicate that all following are not flags?

Reply:
I actually forgot about that convention. I'm not sure I want to support it though. I'd rather things always match https://github.com/google/closure-compiler/wiki/Flags-and-Options